File: frontend/src/services/orderService.js
import { api } from './api'

export const orderService = {
  createOrder: (data) => api.post('/orders', data),

  getOrders: (page = 0, size = 10) =>
    api.get(`/orders?page=${page}&size=${size}`),

  getOrderById: (orderId) => api.get(`/orders/${orderId}`),

  getSellerOrders: (page = 0, size = 10) =>
    api.get(`/orders/seller?page=${page}&size=${size}`),
}

File: order-service/src/main/java/org/n11bootcamp/orderservice/services/OrderService.java
package org.n11bootcamp.orderservice.services;



import org.n11bootcamp.orderservice.dtos.requests.CreateOrderRequest;
import org.n11bootcamp.orderservice.dtos.responses.OrderResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.UUID;

public interface OrderService {

    OrderResponse createOrder(UUID userId, CreateOrderRequest request);

    OrderResponse getOrder(UUID orderId, UUID userId);

    Page<OrderResponse> getOrders(UUID userId, Pageable pageable);

    Page<OrderResponse> getOrdersBySeller(UUID sellerId, Pageable pageable);
}
